test/constraints-validation.test.ts

```diff
@@ -0,0 +1,190 @@
+  for (let i = 0; i < array.length; i++) {
+    array[i] = { id: "" + i };
+  }
+  return array;
+}
+
+function singleErrorInArray(id: string, errorNumberVariable?: number) {
+  return new blueprints.GenericClusterProvider({
+    version: KubernetesVersion.V1_21,
+    autoscalingNodeGroups: [
+      {
+        id: id + 1
+      },
+      {
+        id: id + 2
+      },
+      {
+        id: id + 3
+      },
+      {
+        id: id + 4,
+        minSize: errorNumberVariable
+      }
+    ]
+  });
+}
+
+function createFargateProfile(fargateProfileName: string) {
+  return new blueprints.GenericClusterProvider({
+    version: KubernetesVersion.V1_21,
+    fargateProfiles: {
+      "fp1": {
+        fargateProfileName: fargateProfileName,
+        selectors: [{ namespace: "serverless1" }]
+      }
+    }
+  });
+}
+function getConstraintsDataSet(): DataError[] {
+
+  let result: [Executable, ZodError][] = [];
+
+  const blueprint1 = blueprintBase.clone().id(longName);
+  const blueprint2 = blueprintBase.clone().enableControlPlaneLogTypes(CONTROL_PLANE_LOG_TYPE.authenticator).name("").id("idName");
+
+  //be sure all blueprint tests built have a id defined before this!
+  result.push([() => blueprint1.build(app, blueprint1.props.id!), tooBigString63]);
+  result.push([() => blueprint2.build(app, blueprint2.props.id!), tooSmallString1]);
+
+  result.push([() => createFargateProfile(longName), tooBigString63]);
+  result.push([() => createFargateProfile(""), tooSmallString1]);
+
+  result.push([() => createAutoScalingGroup(longName), tooBigString63]);
+  result.push([() => createAutoScalingGroup(""), tooSmallString1]);
+  result.push([() => createAutoScalingGroup("Name", 6000), tooBigNumber5000]);
+  result.push([() => createAutoScalingGroup("Name", -1), tooSmallNumber0]);
+
+  result.push([() => new blueprints.addons.MetricsServerAddOn({ name: longName }), tooBigString63]);
+  result.push([() => new blueprints.addons.MetricsServerAddOn({ repository: "improper.website" }), notUrlError]);
+
+  result.push([() => createManyAutoScalingGroup(6000), tooBigNumber5000]);
+  result.push([() => singleErrorInArray("Name", 6000), tooBigNumber5000]);
+
+  return result;
+}
+
+function compareIssues(object1: any, object2: any) {
+  const keys1 = Object.keys(object1);
+  const keys2 = Object.keys(object2);
+
+  if (keys1.length !== keys2.length) {
+    return false;
+  }
+  for (let key of keys1) {
+    if (key == "message" || key == "path" || key == "inclusive") {
+      continue;
+    }
+
+    if (object1[key] !== object2[key]) {
+      return false;
+    }
+  }
+  return true;
+}
+
+test("So given that specific validation constraints have been done prior, when creating EKS Blueprints, ClusterProviders, and Helm Add-ons with invalid attributes, the tests will fail with the expected errors.", () => {
+  getConstraintsDataSet().forEach((ex) => {
+    try {
+      ex[0]();
+      expect(true).toBe(false);
+    } catch (e) {
+      console.log(e);
```

Comment:
Do we need this console.log?

Reply:
No I forgot to remove that and I noticed it right after I pushed it.

Reply:
Of course, you can commit as many times as you want.  Once you are ready for review, in the PR Reviewers section, click the "Re-request review" button in front of my name.